htdocs/projet/class/project.class.php

```diff
@@ -1322,5 +1323,74 @@ function update_element($TableName, $ElementSelectId)
 		}
 
 	}
+
+    /**
+     * Is the project having a delay?
+     * Requires: $this->statut and $this->date_end to be defined.
+     * Also it requires $conf->projet->warning_delay to be set
+     *
+     * @return bool
+     */
+    public function hasDelay()
+    {
+        global $conf;
+
+        if (!$this->date_end || !$conf->projet->warning_delay) {
+            return false;
+        }
+
+        $now = dol_now();
+
+        //Open status
+        if ($this->statut == 1 && ($this->date_end < ($now - $conf->projet->warning_delay))) {
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     *      Load indicators for dashboard (this->nbtodo and this->nbtodolate)
+     *
+     *      @param	User	$user        		Objet user
+     *      @return int         				<0 if KO, 0=Nothing to show, >0 if OK
+     */
+    public function load_board(User $user)
```

Comment:
To make Travis (and me) quite happy, you could implement this into loadBoard() and keep this method as a deprecated wrapper.